src/docgen/render/markdown.ts

```diff
@@ -180,21 +181,8 @@ export class Markdown {
 
     const content: string[] = [];
     if (this.header) {
-      const anchor = anchorForId(this.id ?? '');
       const heading = `${'#'.repeat(headerSize)} ${this.header}`;
-
-      // This is nasty, i'm aware.
-      // Its just an escape hatch so that produce working links by default, but also support producing the links that construct-hub currently relies on.
-      // This will be gone soon.
-      // Note though that cross links (i.e links dependencies will not work yet regardless)
-      const headerSpan = !!process.env.HEADER_SPAN;
-      if (headerSpan) {
-        content.push(
-          `${heading} <span data-heading-title="${this.header}" data-heading-id="${anchor}"></span>`,
-        );
-      } else {
-        content.push(`${heading} <a name="${this.id}" id="${anchor}"></a>`);
-      }
+      content.push(`${heading} <a id="${this.id}"></a>`);
```

Comment:
Currently the id's that are generated in the schema are the full JSII FQNs - e.g. `@aws-cdk/aws-s3.BucketProps.parameter.accessControl`. But as mentioned in https://github.com/cdklabs/construct-hub-webapp/issues/714, it would be nice to just use assembly-relative IDs for URL fragments in Construct Hub (i.e. just `BucketProps.parameter.accessControl`).

For example, our current URLs look like:
`https://constructs.dev/packages/@aws-cdk/aws-s3/v/1.137.0/api/softwareamazonawscdkservicess3bucket#softwareamazonawscdkservicess3bucketpropsparameterbucketname`
and with the current state of the PR, we can probably make it
`https://constructs.dev/packages/@aws-cdk/aws-s3/v/1.137.0/api/Bucket#@aws-cdk/aws-s3.BucketProps.parameter.bucketName`
but we could optimize it further and make it
`https://constructs.dev/packages/@aws-cdk/aws-s3/v/1.137.0/api/Bucket#BucketProps.parameter.bucketName`

That is probably what we want at the end of the day, so I need to update the IDs used by markdown rendering to be customizable. TODO